### PR TITLE
Contribution by filipczekajlo

### PR DESCRIPTION
**Pull request by filipczekajlo**

### CODE REVIEW

File: color/p5js/colm.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     col.c3 = 64;
+   }
+   
+   if(p2==372){
+     col.c1 = 220;
+     col.c2 = 40;
+     col.c3 = 33;
+   }
+   
+  if (p2>372){ // <--How do I make those magic numbers disappear so that I can add offset or a while loop so that it changes the color when it loops?
```
Comment (fabianmoronzirfas) [inline color/p5js/colm.js]: If you are not sure how your example should work you should not make a PR out of it.

File: color/p5js/colm.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+col = {
```
Comment (fabianmoronzirfas) [inline color/p5js/colm.js]: The file should be named index.js not colm.js

File: color/p5js/colm.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  if (p2>372){ // <--How do I make those magic numbers disappear so that I can add offset or a while loop so that it changes the color when it loops?
+    background(57,9,85);
+   // background(57,9,85);
+    p2= 0;
+  }
+  
+
+}
+function mousePressed () {
+save ('image.jpg');
```
Comment (fabianmoronzirfas) [inline color/p5js/colm.js]: Please remove the image saving for the site.

File: color/p5js/index.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---  
+layout: p5js
+author: [Filip Czekajlo]
+author-url: [-]
+title: [colm]
+tags: [P5.js,[Color], [-], ]
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: no inner `[]` please only the outer one.  See for example https://raw.githubusercontent.com/fabianmoronzirfas/gestalten-in-code/master/grid/p5js/corridor/index.md

File: color/p5js/index.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+permalink:   gestalten-in-code/color/p5js/colm.js
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: permalink should be something like `color/colm` not the name to a file

File: color/p5js/index.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---  
+layout: p5js
+author: [Filip Czekajlo]
+author-url: [-]
+title: [colm]
+tags: [P5.js,[Color], [-], ]
+permalink:   gestalten-in-code/color/p5js/colm.js  
+summary: [-]
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: Please add a summary.

File: color/p5js/index.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---  
+layout: p5js
+author: [Filip Czekajlo]
+author-url: [-]
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: don't leave this out

File: color/p5js/index.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+---  
+layout: p5js
+author: [Filip Czekajlo]
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: Don't leave this out

File: color/p5js/index.md (hunk 4 above)
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: What means [-] remove it if you have no idea which law it is.

File: color/p5js/index.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+author-url: [-]
+title: [colm]
+tags: [P5.js,[Color], [-], ]
+permalink:   gestalten-in-code/color/p5js/colm.js  
+summary: [-]
+---  
+
+
+
+![]( gestalten-in-code/color/p5js/out.png )
```
Comment (fabianmoronzirfas) [inline color/p5js/index.md]: use relativ paths for images. The code from your index.js will be added to the page. So there is no need for the image.  

A relativ path would be

File: color/p5js/colm.js (hunk 2 above)
Comment (fabianmoronzirfas) [inline color/p5js/colm.js]: This is still wrong

**Comment by filipczekajlo** [comment]
Hey, 
I just updated my files. I hope I did everything right :)

File: color/p5js/colm/index.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Filip Czekajlo
+author-url: https://github.com/filipczekajlo
+title: colm
+tags: [P5.js, color]
```
Comment (fabianmoronzirfas) [inline color/p5js/colm/index.md]: there is no law of gestalt where this could be added?

File: color/p5js/colm/index.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+permalink: /color/colm
+summary: Linear changing color gradients.
```
Comment (fabianmoronzirfas) [inline color/p5js/colm/index.md]: This is very short.

File: motion/p5js/green-hyperspace/index.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
+var lines = [];
+
+
+function setup() {
+  createCanvas(500, 500);
+  for (var l = 0; l < 650; l++) {
+    lines[l] = {
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.js]: Making functions in a loop is a bad design. You should have a function that creates one of your line objects and then call it in the iteration.  

I suggest a design like this:  

```js
var lines = [];

function aline () {
  this.x = random(0, width);
  this.y = random(0, height);
  this.display = function display () {
    stroke(random(125), random(250), random(125), 255);
    line(height/2, width/2, this.x, this.y);
  };
  this.move = function move() {
    this.x = this.x + random(-3, 3);
    this.y = this.y + random(-3, 3);
  }
}
function setup() {
  createCanvas(500, 500);
  for (var l = 0; l < 650; l++) {
    lines[l] = new aline();
  }
}

function draw() {
  background(0);
  for (var l = 0; l < lines.length; l++) {
    lines[l].move();
    lines[l].display();
  }
}

```

File: motion/p5js/green-hyperspace/index.md [hunk 13]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Filip Czekajlo
+author-url: https://github.com/filipczekajlo
+title: green-hyperspace
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.md]: No hypens. This is a title and should be `Title Case`

File: motion/p5js/green-hyperspace/index.md [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+tags: [P5.js, motion]
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.md]: no law of gestalt?

File: motion/p5js/green-hyperspace/index.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+permalink: /motion/green-hyperspace
+summary: Travelling at light speed through hyperspace.
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.md]: hm. This is a wired description.

File: projects/Herbert-Bayer/index.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+authors-url: 
+  - "https://github.com/filipczekajlo"
+layout: project
+title: "Herbert Bayer - deconstructing master artists"
+type: project
+authors:
+  - "Filip Czekajlo"  
+---
+## Deconstructing Master Artists
+![](./splash.png)
```
Comment (fabianmoronzirfas) [inline projects/Herbert-Bayer/index.md]: You have no image called splash.png

Comment (fabianmoronzirfas) [inline projects/Herbert-Bayer/index.md]: Oh sorry you have.

File: projects/fractal-lines/index.md [hunk 17]
```diff
@@ -0,0 +1,97 @@
+--- 
+authors: 
+  - "Filip Czekajlo"
+authors-url: 
+  - "https://github.com/filipczekajlo"
+layout: Project
+title: "Fractal-Lines"
```
Comment (fabianmoronzirfas) [inline projects/fractal-lines/index.md]: Does it need the hyphen?

**Comment by fabianmoronzirfas** [comment]
@filipczekajlo are you done? can I review it?

**Comment by fabianmoronzirfas** [comment]
ping?

**Comment by filipczekajlo** [comment]
Hey I just updated it again. Hope you can see it now. If not, I will close this pull request and make a new one.

File: motion/p5js/green-hyperspace/index.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    stroke(random(125), random(250), random(125), 255);
+    line(height/2, width/2, this.x, this.y);
+  };
+  this.move = function move() {
+    this.x = this.x + random(-3, 3);
+    this.y = this.y + random(-3, 3);
+  }
+}
+function setup() {
+  createCanvas(500, 500);
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.js]: Please add the canvas assignment as stated in the wiki https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#code

    var canvas = createCanvas(500, 500); // !this is important
    canvas.parent('sketch'); // !this as well

We need this to display the sketch in the page

File: color/p5js/colm/index.md [hunk 19]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Filip Czekajlo
+author-url: https://github.com/filipczekajlo
+title: colm
+tags: [P5.js, color, Law of Good Gestalt]
```
Comment (fabianmoronzirfas) [inline color/p5js/colm/index.md]: Please add the tags as a list in the frontmatter  

https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#markdown-and-front-matter  

tags: 
  - P5.js
  - color
  - Law of Good Gestalt

File: color/p5js/colm/index.md [hunk 20]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Filip Czekajlo
+author-url: https://github.com/filipczekajlo
+title: colm
```
Comment (fabianmoronzirfas) [inline color/p5js/colm/index.md]: Please make the title TitleCase

title: Colm

File: motion/p5js/green-hyperspace/index.md [hunk 21]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Filip Czekajlo
+author-url: https://github.com/filipczekajlo
+title: Green Hyperspace
+tags: [P5.js, motion]
+permalink: /motion/Green Hyperspace
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.md]: The permalink needs to be URL safe. Make it:  

    permalink: /motion/green-hyperspace

File: motion/p5js/green-hyperspace/index.md [hunk 22]
```diff
@@ -0,0 +1,9 @@
...
+summary: Travelling at light speed through hyperspace.
```
Comment (fabianmoronzirfas) [inline motion/p5js/green-hyperspace/index.md]: This is a bit wired description… Please write something that is more useful

File: projects/fractal-lines/index.md [hunk 23]
```diff
@@ -0,0 +1,97 @@
+--- 
+authors: 
+  - "Filip Czekajlo"
+authors-url: 
+  - "https://github.com/filipczekajlo"
+layout: Project
+title: "Fractal Lines"
+type: Project
```
Comment (fabianmoronzirfas) [inline projects/fractal-lines/index.md]: type: project

File: projects/fractal-lines/index.md [hunk 24]
```diff
@@ -0,0 +1,97 @@
+--- 
+authors: 
+  - "Filip Czekajlo"
+authors-url: 
+  - "https://github.com/filipczekajlo"
+layout: Project
```
Comment (fabianmoronzirfas) [inline projects/fractal-lines/index.md]: layout: project

Comment (fabianmoronzirfas) [inline projects/fractal-lines/index.md]: needs to be lowercase

**Comment by filipczekajlo** [comment]
Ok I fixed everything again.

**Comment by fabianmoronzirfas** [comment]
The image 
    
     

Does not exist

**Comment by fabianmoronzirfas** [comment]
merged and live 

- https://interface.fh-potsdam.de/gestalten-in-code/projects/Herbert-Bayer/
- https://interface.fh-potsdam.de/gestalten-in-code/projects/fractal-lines/
- https://interface.fh-potsdam.de/gestalten-in-code/motion/green-hyperspace/
- https://interface.fh-potsdam.de/gestalten-in-code/color/colm/

**Pull request merged**